frontend/src/components/test-history.js

```diff
@@ -33,334 +31,317 @@ import {
+        ...time_filter,
+        ...env_filter
+      }
+    );
 
-  setPage = (_event, pageNumber) => {
-    this.setState({page: pageNumber}, () => {
-      this.getResultsForTable();
-    });
-  };
+  }, [testResult, filters]);
 
-  pageSizeSelect = (_event, perPage) => {
-    this.setState({pageSize: perPage}, () => {
-      this.getResultsForTable();
-    });
+  const onCollapse = (event, rowIndex, isOpen) => {
+    // lazy-load the result view so we don't have to make a bunch of artifact requests
+    // TODO with ResultView moving tab rendering and artifact fetching into ArtifactTab, this may not be necessary anymore
+    setRows(rows.map((row, index) => {
+      if (index === (rowIndex + 1)){
+        return ({
+          ...row,
+          'cells': [{
+            title: <ResultView defaultTab='summary' hideTestHistory={true} hideSummary={false} hideTestObject={false} testResult={rows[rowIndex].result}/>
+          }]});
+      } else if (index === rowIndex) {
+        return({
+          ...row,
+          'isOpen': isOpen
+        });
+      } else {
+        return(row);
+      }
+    }));
   };
 
-  updateFilters (name, operator, value, callback) {
-    let filters = this.state.filters;
+  const updateFilters = useCallback((name, operator, value) => {
+    let updatedfilters = {...filtersState};
     if ((value === null) || (value.length === 0)) {
-      delete filters[name];
+      delete updatedfilters[name];
     }
     else {
-      filters[name] = {'op': operator, 'val': value};
+      updatedfilters[name] = {'op': operator, 'val': value};
     }
-    this.setState({filters: filters, page: 1}, callback);
-  }
+    setFiltersState(updatedfilters);
+    setPage(1);
+  }, [filtersState]);
 
-  setFilter = (field, value) => {
+  const setFilter = useCallback((field, value) => {
     // maybe process values array to string format here instead of expecting caller to do it?
     let operator = (value.includes(';')) ? 'in' : 'eq';
-    this.updateFilters(field, operator, value, this.refreshResults);
-  };
+    updateFilters(field, operator, value);
+  }, [updateFilters]);
 
-  removeFilter = id => {
-    if ((id !== 'result') && (id !== 'test_id')) {   // Don't allow removal of error/failure filter
-      this.updateFilters(id, null, null, this.refreshResults);
-    }
-  };
-
-  onFailuresCheck = (checked) => {
-    let { filters } = this.state;
-    filters['result']['val'] = ('failed;error') + ((checked) ? ';skipped;xfailed' : ';skipped;xfailed;xpassed;passed');
-    this.setState(
-      {onlyFailures: checked, filters},
-      this.refreshResults
-    );
-  };
+  useEffect(()=>{
+    if (comparisonResults !== undefined) {
+      // setResults(comparisonResultsState)
+      setRows(comparisonResults.map((result, index) => resultToTestHistoryRow(result, index, setFilter)).flat());
+    } else {
+      setRows([getSpinnerRow(4)]);
+      setIsEmpty(false);
+      setIsError(false);
+      let params = buildParams(filtersState);
```

Comment:
Nitpick: Could be const since we are not re-assigning object but changing it's props